types/foundry/client/canvas/layers/base/placeables-layer.d.mts

```diff
@@ -381,7 +381,9 @@ export default class PlaceablesLayer<TObject extends PlaceableObject = Placeable
 
     protected override _onDragLeftDrop(event: PlaceablesLayerPointerEvent<TObject>): void;
 
-    protected override _onDragLeftCancel(event: PlaceablesLayerPointerEvent<TObject>): void;
+    protected override _onDragLeftCancel(
+        event: PIXI.FederatedPointerEvent | PlaceablesLayerPointerEvent<TObject>,
+    ): void;
```

Comment:
?

Reply:
That's the v12 defintion. Probably for the same reason as now.
`canvas.stage.on` only takes `PIXI.FederatedPointerEvent`s so the preview listeners can't be `PlaceablesLayerPointerEvent<TObject>` and the call to `TemplateLayerPF2e#_onDragLeftCancel` in `#deactivatePreviewListeners` doesn't take the `PIXI.FederatedPointerEvent`.

Using a type assertion would also work:
```ts
    protected override _onDragLeftCancel(
        event: PIXI.FederatedPointerEvent | PlaceablesLayerPointerEvent<TObject>,
    ): void {
        // Prevent normal drag operations when a preview is active
        if (this.#previewListeners) return;
        return super._onDragLeftCancel(event as PlaceablesLayerPointerEvent<TObject>);
    }
```

Reply:
But a `PlaceablesLayerPointerEvent` _is_ a `PIXI.FederatedPointerEvent`. If a method won't accept a `PIXI.FederatedPointerEvent`, it also won't accept a `PlaceablesLayerPointerEvent`.

Reply:
yeah but typescript complains that `interactionData` could be missing in the `PIXI.FederatedPointerEvent` because it's not optional in `PlaceablesLayerPointerEvent`.
And I don't think the type of `canvas.stage.on` events can be changed as those are internal pixi types I believe.